impress-org/givewp
# Refactor: events as opt-in feature

Resolves [GIVE-438]

## Description



This is a pivot that makes events an opt-in feature for customers who want to intentionally try it out while in beta.

Updates:
- Adds a simple beta features notifications counter and uses it to turn event tickets into an opt-in flow by default instead of opt-out.
- Added a programatic way to enable the events feature with the constant `define('GIVE_FEATURE_ENABLE_EVENT_TICKETS', true)`
- Fixes confirmation page total amount

## Affects



Event tickets & beta features

## Visuals



<img width="856" alt="Screenshot 2024-03-11 at 7 58 32 PM" src="https://github.com/impress-org/givewp/assets/10138447/19585a6a-d5d9-446a-9cdf-2f94aa1e9606">

<img width="1128" alt="Screenshot 2024-03-11 at 7 58 41 PM" src="https://github.com/impress-org/givewp/assets/10138447/685041cf-77bd-495b-8eb2-20cb81d2053d">

<img width="1290" alt="Screenshot 2024-03-11 at 7 58 50 PM" src="https://github.com/impress-org/givewp/assets/10138447/1d2174c2-bdfb-42f2-af2d-797061a711d3">

## Testing Instructions



- Upon first install/update a notification counter should appear in the givewp settings menu item
- Clicking the settings and then beta features tab will complete that notification counter and navigating to another page will make it dissapear
- Events should only show up when set to enabled.

## Pre-review Checklist






[GIVE-438]: https://stellarwp.atlassian.net/browse/GIVE-438?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
